packages/medusa/src/services/order-edit.ts

```diff
@@ -521,7 +526,11 @@ export default class OrderEditService extends TransactionBaseService {
     orderEdit.items.forEach((item) => {
       if (item.adjustments?.length) {
         item.adjustments.forEach((adjustment) => {
-          clonedItemAdjustmentIds.push(adjustment.id)
+          if (
+            config.preserveCustomAdjustments ? !!adjustment.discount_id : true
```

Comment:
nit: Move that to a const, it looks weird to have a ternary expression in the if